server/src/main/java/org/elasticsearch/index/IndexModule.java

```diff
@@ -349,13 +349,13 @@ public boolean match(String setting) {
     }
 
     /**
-     * Factory for creating new {@link IndexSearcherWrapper} instances
+     * Factory for creating new {@link IndexReaderWrapper} instances
      */
-    public interface IndexSearcherWrapperFactory {
+    public interface IndexReaderWrapperFactory {
```

Comment:
do we need this or can we break it even more and just use `Function<IndexService, IndexReaderWrapper>`  Eventually I'd love to remove this indirection altogether and maybe just have an wrapper function but that's a followup

Reply:
I removed this class in https://github.com/elastic/elasticsearch/pull/43645/commits/615c48b31403c587880f5beaff871e7eae73901f